schema.go

```diff
@@ -44,46 +44,23 @@ type serializationInfo struct {
 	resources map[reflect.Type]string
 	types     map[reflect.Type]string
 	enums     map[reflect.Type]string
-	inputMap  inputToImplementor
+	inputMap  inputmap.InputToImplementor
 }
 
 // Serialize a package to JSON Schema.
 func serialize(opts options) (string, error) {
-	pkgSpec, err := serializeSchema(opts)
-	if err != nil {
-		return "", err
-	}
-	pkgSpec.Language = opts.Language
-
-	schemaJSON, err := json.MarshalIndent(pkgSpec, "", "  ")
-	if err != nil {
-		return "", err
-	}
-
-	return string(schemaJSON), nil
-}
-
-// Get the packagespec given resources, etc.
-func serializeSchema(opts options) (schema.PackageSpec, error) {
-	spec := schema.PackageSpec{}
-	spec.Resources = make(map[string]schema.ResourceSpec)
-	spec.Types = make(map[string]schema.ComplexTypeSpec)
-	spec.Name = opts.Name
-	spec.Version = opts.Version.String()
-
 	info := serializationInfo{}
 	info.pkgname = opts.Name
 	info.resources = make(map[reflect.Type]string)
 	info.types = make(map[reflect.Type]string)
 	info.enums = make(map[reflect.Type]string)
-	info.inputMap = initializeInputMap()
+	info.inputMap = inputmap.GetInputMap()
```

Comment:
It's idiomatic to construct this in place, instead of creating the zero initialized struct and adding member fields.